frappe/model/base_document.py

```diff
@@ -1058,6 +1075,57 @@ def _extract_images_from_text_editor(self):
 			for df in self.meta.get("fields", {"fieldtype": ('=', "Text Editor")}):
 				extract_images_from_doc(self, df.fieldname)
 
+	def _cast_date_and_time_fields(self) -> None:
+		"""
+		Converts datetime/string value to date and time respectively for Date and Time fields only.
+		This is necessary since the values with which the Document class is initialized can differ.
+
+		For eg: The user initializes document with datetime values for Date and Time field, the framework
+		should parse them into correct format.
+		If the values are in datetime format, preserve it in datetime format
+		If the values are in string format, preserve it in string format
```

Comment:
@hrwX is preserving the type important for the goal of this PR? My expectation would be that a document always parses date/-times that were passed as strings and always provides a date/-time object back.

Reply:
The docstring wasn't updated. Have changed it now.